src/platform/cannonlake/include/platform/platform.h

```diff
@@ -197,8 +197,9 @@ static inline void platform_panic(uint32_t p)
 #define platform_trace_point(__x) \
 	mailbox_sw_reg_write(SRAM_REG_FW_TRACEP, (__x))
 
-extern struct timer *platform_timer;
+#define PLATFORM_XTAL_OSCILLATOR_CLK 24000000
```

Comment:
It doesn't match ssp clock frequencies in clk-map.h (also for other platforms).
Could we potentially use those defines in this ssp_freq array and use the array in our ssp clock calculations?
Then you could just take the highest clock from the array without the need to make calculations based on cardinal clock define.
Also I'm wondering about decoding additional registers settings for cardinal clock from this array. It would need to be something similar like is used for cpu_freq.

Reply:
I've moved this defines to ssp_freq.